src/timeline/TimelineModel.cpp

```diff
@@ -1170,11 +1181,12 @@ TimelineModel::cacheMedia(QString eventId)
         QString suffix = QMimeDatabase().mimeTypeForName(mimeType).preferredSuffix();
 
         const auto url = mxcUrl.toStdString();
+        const auto name = QString(mxcUrl).remove("mxc://");
         QFileInfo filename(QString("%1/media_cache/%2.%3")
                              .arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation))
-                             .arg(QString(mxcUrl).remove("mxc://"))
+                             .arg(name)
                              .arg(suffix));
-        if (QDir::cleanPath(filename.path()) != filename.path()) {
+        if (QDir::cleanPath(name) != name) {
```

Comment:
I'm doing this because I noticed that if CacheLocation is not a canonical path itself, this would wrongly mark the url as unsafe.

Reply:
Mhm, that makes sense. Didn't consider that!